### ISSUE TITLE
Docs provide invalid instructions for `nodeType` node

### ISSUE THREAD

**Issue by polarathene** [graysonhicks/gatsby-plugin-remote-images#46]
I tried to follow [this plugins docs](https://www.gatsbyjs.org/packages/gatsby-plugin-remote-images/), which details the config option like so:

```
// The node type that has the images you want to grab.
// This is generally the camelcased version of the word
// after the 'all' in GraphQL ie. allMyImages type is myImages
nodeType: 'myNodes',
```

### Example Data

`gatsby-config.js` snippet:
```js
`gatsby-transformer-yaml`,
{
  resolve: `gatsby-source-filesystem`,
  options: {
    path: `${__dirname}/data`,
  },
},
{
  resolve: `gatsby-plugin-remote-images`,
  options: {
    nodeType: 'pugsYaml',
    imagePath: 'nodes[].url',
  },
},
```

query example:
```graphql
query {
  images: allPugsYaml(filter: {image_type: {eq: "tall"}}) {
    nodes {
      localImage {
        childImageSharp {
          fluid(maxWidth: 500) {
            ...GatsbyImageSharpFluid_withWebp
          }
        }
      }
    }
  }
}
```

`data/pugs.yml`:
```yaml
- url: https://images.unsplash.com/photo-1512723185835-0700e5069a9a
  image_type: tall
  image_id: 0
- url: https://images.unsplash.com/photo-1517423440428-a5a00ad493e8
  image_type: big
  image_id: 1
- url: https://images.unsplash.com/photo-1505917812101-1a9d82610cc2
  image_type: wide
  image_id: 2
```

### Problem

According to the docs `allPugsYaml` with it's `pugsYaml` nodes should mean this plugin is configured with ` nodeType: 'pugsYaml'` as shown above, but this fails with:

```
There was an error in your GraphQL query:

Cannot query field "localImage" on type "PugsYaml".
```

### Solution

Updating the plugin config to reflect that error of the type mentioned `nodeType: 'PugsYaml'`, fixed that error. 

---

### Troubleshooting

Below can be ignored, besides the `imagePath` misunderstanding, it's not relevant to the issue.

However, now I've run into an error others have been having (querying for any other property like the url works fine though):

```
Cannot read property 'localImage' of undefined
```

Inspecting in GraphiQL shows the type listed in my `PugsYaml` node(it's `pugsYaml` in the left sidebar list) with all the fields expected for `localImage`, but querying (without the sharp fragment, just any field for localImage) fails too. Same error as above.

---

**UPDATE:** This was a misunderstanding of the docs example about arrays where it mentions `imagePath: 'nodes[].imageUrl'`. That `nodes[]` part was unrelated to my query on `nodes` which returns an array of `PugsYaml` nodes.. the docs were referring to the graphql response providing the url for each image in a nested level of each image node, not the array of the nodes themselves..

`imagePath: 'url'` resolved that issue, where the images would actually download now. This wasn't successful however, but the error is unrelated to the plugin afaik:

```
gatsby-plugin-remote-images ERROR: failed to process https://images.unsplash.com/photo-1511431833882-509a76113392
TimeoutError: Timeout awaiting 'request' for 30000ms
```

**UPDATE 2:** Timeout issues were resolved via an [active PR](https://github.com/gatsbyjs/gatsby/pull/24899), then hit retry errors failing due to poor network connection speeds/quality, `GATSBY_CONCURRENT_DOWNLOAD=2` env var limits the amount of concurrent downloads, allowing for a successful build.

**Comment by polarathene** [comment]
My own issues aside, the **TL;DR:** version regarding the title is that the docs describe `nodeType` should be camel cased, but it seems that it should be Pascal Type (which the GraphQL node type uses)

**Comment by PatrykBuniX** [comment]
Yeah, I've had the same problem. And using pascal case fixed it. 
This one works for me: `nodeType: "MyNodes"`, while this one does not: `nodeType: "MyNodes"`

**Comment by graysonhicks** [comment]
Hm, I have seen this work both ways, but I think y'all are right overall.  The naming convention for the internal `type` of a node is up to the source plugin for those nodes.  If you write your own source plugin you may set the `type` as `myNodes` and the `nodeType: 'myNodes'` would work.  This could also be the case if you are using a third party source plugin doing the same.

That being said, looking at the docs here: https://www.gatsbyjs.org/tutorial/source-plugin-tutorial/#create-nodes-inside-of-sourcenodes-with-the-createnode-function, it does use the pascal case `Post` as the type (vs. `post`).  

So I'll update the docs with pascal case and hope no one opens an issue saying it should be camel case later. 😅 

Thanks @polarathene and @PatrykBuniX !

**Comment by graysonhicks** [comment]
Addressed by #51

**Comment by polarathene** [comment]
Probably could mention that it varies based on what plugin is naming the node type as. If it's not clear to the user what the plugin is using, they could try the advised casing, then because the docs mention the casing might differ, they can try that next instead of wondering why it's not working :sweat_smile: 

For me my nodes were from the source plugins for json and yaml, which I thought were pretty official/common.

**Comment by graysonhicks** [comment]
Released with v2.2.0
